### PR TITLE
Manage Properties Model

### LINKED ISSUE #37: Manage Properties Model
Like we have with DOMDocument in LSP4XML, we should have a PropertiesModel which stores offsets of 

 * key, value of each properties
 * comments

It will clean code for any services like:

 * completion
 * hover
 * validation
 * symbols
 * folding

### PR DESCRIPTION
**Pull request by angelozerr**
Fixes #37

This PR provides a Properties Model which is able to manage start/end offset for:

 * comments
 * property key
 * delemiter assign
 * property value

We can compare this properties model like a DOM document for XML. This properties model is consumed in the Quarkus Language Server for completion and hover. 

It could be used in another services like diagnostics, symbols, higlight, etc

You can find test with parser in PropertiesModelTest. 

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
@fbricon please review this PR and merge it because @xorye need the properties model for the validation issue https://github.com/redhat-developer/quarkus-ls/pull/36

@xorye I think parser should be improved again to support some feature like

`a = b # comments`

Please create issue with limitation once this PR will be merged. If you can fix it yourself, it should be cool otherwise I will try to do it when you will create issues. Please tell me what you wish to do.

**Comment by xorye** [comment]
> Please create issue with limitation once this PR will be merged. If you can fix it yourself, it should be cool otherwise I will try to do it when you will create issues. Please tell me what you wish to do.

@angelozerr I would be excited to fix that myself after this is merged.

**Comment by fbricon** [comment]
`b # comments` is a's value. There's nothing else to do

**Comment by xorye** [comment]
does that mean `# comments` is not considered a comment?

**Comment by fbricon** [comment]
yes, it's a part of the value

**Comment by fbricon** [comment]
the fact the syntax highlighter colours it differently is a bug

File: quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/parser/PropertiesModelTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		String text = " a=";
+		PropertiesModel model = PropertiesModel.parse(text, "application.properties");
+		Assert.assertEquals(model.getChildren().size(), 1);
+
+		Node firstPropertyNode = model.getChildren().get(0);
+		assertProperty(firstPropertyNode, 1, 2, "a", 3, -1, -1, null);
+
+	}
+	
+	private static void assertComents(Node comments, int expectedStart, int expectedEnd, String expectedText) {
```
Comment (fbricon) [inline quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/parser/PropertiesModelTest.java]: assertComments

**Comment by fbricon** [comment]
@angelozerr  please rebase and merge

File: quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/parser/PropertiesModelTest.java (hunk 1 above)
Comment (angelozerr) [reply quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/parser/PropertiesModelTest.java]: fixed

**Pull request merged**